### ISSUE TITLE
look for comment directives at *end* of in-line comment, allow other in-line commentary

### ISSUE THREAD

**Issue by jtmoon79** [python/mypy#8225]
**tl;dr** mypy should process directives with other in-line commentary, e.g. `foo = None  # make sure to bar this # type: ignore` should be processed by mypy to ignore this line.

### mypy could co-exist with other in-line comments if it checked the _end_ of the in-line comment string

For example, the following `file1.py` causes an error for mypy using Python 3.5

```Python
from typing import NamedTuple
C = NamedTuple('C', [('a', str), ('b', str)])
C.__new__.__defaults__ = (None,  # a
                          None,  # b
                          )
```

Running `python -m mypy file1.py` results in

```
error: "Callable[[Type[C], VarArg(Any), KwArg(Any)], Any]" has no attribute "__defaults__"
```

(this not-recommended default values declaration is to allow running in Python 3.5).

<br />

I appended a `# type: ignore` comment to the in-line comment, e.g.
```Python
from typing import NamedTuple
C = NamedTuple('C', [('a', str), ('b', str)])
C.__new__.__defaults__ = (None,  # a  # type: ignore
                          None,  # b
                          )
```

However, running `mypy` still fails with the same error.  The workaround is to re-align the comments like:

```Python
from typing import NamedTuple
C = NamedTuple('C', [('a', str), ('b', str)])
C.__new__.__defaults__ = (# type: ignore
                          None,  # a
                          None,  # b
                          )
```

This is a good workaround; _works-as-expected_.

<br />

_However_, mypy could give a little leeway to users if it checked the _end_ of the in-line comment for the special directives, i.e. `comment.endswith('# type: ignore')`.

<br />

----

Tested using Python 3.5.3 and mypy 0.750.

**Comment by ethanhs** [comment]
mypy will read type comments *before* other comments. Otherwise it would be impossible to comment out a type comment.
